grails-app/init/org/pih/warehouse/BootStrap.groovy

```diff
@@ -605,8 +607,18 @@ class BootStrap {
             liquibase = new Liquibase('views/drop-all-views.xml', new ClassLoaderResourceAccessor(), database)
             liquibase.update(null as Contexts, new LabelExpression());
 
-            // FIXME Not good that we'll need to update this on subsequent versions so this needs some more thought
-            List previousChangelogVersions = ["0.5.x", "0.6.x", "0.7.x", "0.8.x"]
+            // Get all directories from /migrations
+            PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver()
+            Resource[] resources = resolver.getResources("file:grails-app/migrations/*")
+
+            // Find directories with names matching current versions pattern
+            List<String> changelogVersions = resources
+                    .collect { it.filename }
+                    .findAll { it.matches("[0-9]{1,3}.[0-9]{1,3}.x") }
```

Comment:
Can you also add a comment to describe the regexp being used? Something like 
```
// Find directories with names matching current versions pattern which 
// will match to any version number between 0.0.x to 999.999.x
```
Also let's replace [0-9] with 
```
.findAll { it.matches("[[:digit:]]{1,3}.[[:digit:]]{1,3}.x") }
```

Reply:
It's interesting, because `[[:digit:]]` seems to be not working :thinking: 
![image](https://github.com/openboxes/openboxes/assets/83239466/4d5dbb5f-d610-4a89-b818-1c331aa2ac67)
neither with single `[`:
![image](https://github.com/openboxes/openboxes/assets/83239466/c2b1817a-2800-4a48-9a04-af907715ae33)
but it works with \d and just [0-9]:
![image](https://github.com/openboxes/openboxes/assets/83239466/2a7c68ca-9866-4a9e-a09f-23586934635d)

Reply:
Let's use \\d then.